frontend/packages/knative-plugin/src/catalog/useEventSourceProvider.ts

```diff
@@ -0,0 +1,101 @@
+import { CamelKameletBindingModel, CamelKameletModel } from '../models';
+import { getEventSourceIcon } from '../utils/get-knative-icon';
+import { getEventSourceCatalogProviderData } from './event-source-data';
+
+const normalizeEventSources = (
+  eventSources: K8sKind[],
+  namespace: string,
+  t: TFunction,
+): CatalogItem[] => {
+  const normalizedEventSources: CatalogItem[] = eventSources.map((eventSource) => {
+    const { kind, label: name, id: uid } = eventSource;
+    const href = `/catalog/ns/${namespace}/eventsource?sourceKind=${kind}`;
+    const { description, provider } = getEventSourceCatalogProviderData(kind, t) ?? {};
+    return {
+      uid,
+      name,
+      description,
+      icon: { url: getEventSourceIcon(referenceForModel(eventSource)), class: null },
+      type: 'EventSource',
+      provider,
+      cta: { label: t('knative-plugin~Create Event Source'), href },
+    };
+  });
+  return normalizedEventSources;
+};
+
+const normalizeKamelets = (
+  kamelets: K8sResourceKind[],
+  namespace: string,
+  t: TFunction,
+): CatalogItem[] => {
+  const normalizedKamelets = kamelets.map((k) => {
+    const {
+      kind,
+      metadata: { uid, name, creationTimestamp },
+      spec: {
+        definition: { title, description },
+      },
+    } = k;
+    const iconUrl = getEventSourceIcon(kind, k);
+    const href = `/catalog/ns/${namespace}/eventsource?sourceKind=${CamelKameletBindingModel.kind}&name=${name}`;
+    return {
+      uid,
+      name: title,
+      description,
+      creationTimestamp,
+      cta: { label: t('knative-plugin~Create Event Source'), href },
+      type: 'EventSource',
+      icon: { url: iconUrl },
+    };
+  });
+  return normalizedKamelets;
+};
+
+const useKamelets = (namespace: string): [K8sResourceKind[], boolean, any] => {
+  const k8sModel = useSelector(({ k8s }) =>
+    k8s.getIn(['RESOURCES', 'models', referenceForModel(CamelKameletModel)]),
+  );
+  const resource: WatchK8sResource = React.useMemo(
+    () => ({ kind: referenceForModel(CamelKameletModel), isList: true, namespace, optional: true }),
+    [namespace],
+  );
+  const [kamelets, kameletsLoaded, kameletsLoadError] = useK8sWatchResource<K8sResourceKind[]>(
+    resource,
+  );
+  return React.useMemo(
+    () => (!k8sModel ? [[], true, undefined] : [kamelets, kameletsLoaded, kameletsLoadError]),
+    [k8sModel, kamelets, kameletsLoadError, kameletsLoaded],
+  );
+};
+
+const useEventSourceProvider: CatalogExtensionHook<CatalogItem[]> = ({
+  namespace,
+}): [CatalogItem[], boolean, any] => {
+  const { t } = useTranslation();
+  const { loaded, eventSourceModels } = useEventSourceModels();
+  const [kamelets, kameletsLoaded, kameletsLoadError] = useKamelets(namespace);
+  const normalizedSources = React.useMemo(
+    () => [
+      ...(loaded ? normalizeEventSources(eventSourceModels, namespace, t) : []),
+      ...(kameletsLoaded ? normalizeKamelets(kamelets, namespace, t) : []),
```

Comment:
There can be multiple providers for same item type in the catalog. So `useKamelets` and `useEventSources` can be two separate providers providing the item type of `EventSource`. I think it would be better to split the logic here into two providers.

Reply:
Great!! Thanks!! Didn't know that. Providing multiple providers for the same type would resolve this https://github.com/openshift/console/pull/7338#issuecomment-734705818 as well. Currently, I have a workaround in place.

Reply:
fixed